src/configure/clients/modaRepositoryAnalysisClient.ts

```diff
@@ -0,0 +1,27 @@
+import { IRestResponse, RestClient } from "typed-rest-client";
+import vscodeUri from "vscode-uri";
+import { RepositoryAnalysisRequest } from "../model/models";
+import { IRepositoryAnalysisClient } from "./repositoryAnalyisClient";
+
+export class ModaRepositoryAnalysisClient implements IRepositoryAnalysisClient {
+    private restClient: RestClient;
+    private githubPat: string;
+    private pathUrl: string;
+
+    constructor(url: string, githubPat: string) {
+        const u = vscodeUri.parse(url);
+        this.restClient = new RestClient("deploy-to-azure", u.scheme + "://" + u.authority);
+        this.pathUrl = u.path;
+        this.githubPat = githubPat;
+    }
+
+    public async getRepositoryAnalysis(body: RepositoryAnalysisRequest): Promise<IRestResponse<unknown>> {
```

Comment:
why `unknown` type. I guess we have a well defined type for repo analysis result.